modules/oauth-customizing-the-oauth-server-URL.adoc

```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// * authentication/configuring-internal-oauth.adoc
+
+[id="customizing-the-oauth-server-url_{context}"]
+= Customizing oAuth server URL
+
+Custom route functionality for the oAuth server uses the `ingress` config route configuration API. A custom hostname and a TLS certificate can be set using the `spec.componentRoutes` part of the configuration.
+
+[id="customizing-the-oauth-server-route_{context}"]
+== Customizing the oAuth server route
+
+.Prerequisites
+
+* Log in to the cluster as a user with administrative privileges.
+
+.Procedure
+
+* Set the custom hostname and, if needed, set TLS on the ingress config's spec:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Ingress
+metadata:
+  name: cluster
+spec:
+  componentRoutes:
+    - name: oauth-openshift
+      namespace: openshift-authentication
+      hostname: <custom-hostname>
+      servingCertKeyPairSecret:
+        name: <secret-name>
+----
++
+If the domain for the custom hostname suffix does not match the cluster domain suffix, then a secret that contains a TLS certificate and key must exist in the `openshift-config` namespace and must be referenced in the `ingress` config. If the domain for the custom hostname suffix matches the cluster domain suffix, then the secret is optional. The secret must contain `tls.crt` and `tls.key` data keys, where the certificate and key are stored, and both must be in a valid format.
```

Comment:
I think English does not put a comma in front of "then" in conditionals.

Reply:
It may be worth mentioning that there is a [TLS Secret](https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets) designed for exactly this purpose.

Reply:
Removed comma.

Mentioned the TLS Secret.